src/main/java/org/apache/pirk/utils/SystemConfiguration.java

```diff
@@ -160,4 +138,74 @@ public static void resetProperties()
     clearProperties();
     initialize();
   }
+
+  /**
+   * Loads the properties from local properties file in the specified directory
+   * <p>
+   * Only files ending in '.properties' will be loaded
+   */
+  public static void loadPropsFromDir(String dirName)
+  {
+    File dir = new File(dirName);
+    File[] directoryListing = dir.listFiles();
+    if (directoryListing != null)
+    {
+      for (File file : directoryListing)
+      {
+        if (file.getName().endsWith(".properties"))
+        {
+          loadPropsFromFile(file);
+        }
+      }
+    }
+  }
+
+  /**
+   * Loads the properties from the specified file
+   */
+  public static void loadPropsFromFile(File file)
+  {
+    if (file.exists())
+    {
+      try (InputStream stream = new FileInputStream(file);)
+      {
+        logger.info("Loading properties file '" + file.getAbsolutePath() + "'");
+        props.load(stream);
+        stream.close();
```

Comment:
No need to explicitly close() the stream now, the try-with-resources does that for you.